### DISCUSSION TITLE
how to get an example with audio-driver and audio-tools compiled in PlatformIO

### DISCUSSION THREAD

**Discussion by BKneer** [pschatzmann/arduino-audio-driver#30] [Q&A] [answered]
Hi Phil and others,
I've got a ESP32 Audio Kit V2.2 A417 and found your libs to easy implement some software on this board.
I'm using PlatformIO for my other projects and could also import your Libs from Github into my starting project. The easy audio-driver example compiles and could be flashed. I also see some logs on in the console.
But now it comes... for each example I try that includes the audio-tools I get compile errors for missing type definition of  output_device_t/ input_device_t. I could find in your docu how those types are defined and used but couldn't find any header file that defines this files. Is there anything missing in my libdeps or is there something special I have to consider in PlatformIO? In my main.cpp (your example code) I've added the Arduino.h include that is necessary when you use PlatformIO.
Looking to the first compiler errors they pop up immediately when "AudioLibs/AudioBoardStream.h" get parsed, so nothing I could do in main.cpp. 
Thanks for your hints...
Bernhard
![compilerErrors](https://github.com/user-attachments/assets/e8865944-7f11-4c5c-9a66-618f0bc83099)

**Comment by pschatzmann** [comment, marked as answer]
I guess something is wrong with your versions. Double check with this:
```
Processing esp32dev (platform: espressif32; board: esp32dev; framework: arduino)
-------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (6.8.1+sha.de553ce) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.20017.0 (2.0.17) 
 - tool-esptoolpy @ 1.40501.0 (4.5.1) 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 35 compatible libraries
Scanning dependencies...
Dependency Graph
|-- audio-driver @ 0.1.0+sha.8779218
|-- audio-tools @ 0.9.8+sha.9d63174
Building in release mode
Retrieving maximum program size .pio/build/esp32dev/firmware.elf
Checking size .pio/build/esp32dev/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=         ]   9.8% (used 32120 bytes from 327680 bytes)
Flash: [=====     ]  47.0% (used 615913 bytes from 1310720 bytes)
```

I was using the following platformio.ini

```
[env:esp32dev]
platform = espressif32
board = esp32dev
framework = arduino
lib_deps = https://github.com/pschatzmann/arduino-audio-driver.git
           https://github.com/pschatzmann/arduino-audio-tools.git
```

As far as I know PlatformIO is still on the old Arduino 2.0.17 release compared to the latest  3.0.4 in Arduino.

**Comment by BKneer** [comment]
Hello Phil,
wow, that was a fast response... thanks a lot. ;-)
If I compare my input with yours there seems to be not too much difference:
Executing task: C:\Users\Bernhard\.platformio\penv\Scripts\platformio.exe run 

Processing esp32dev (platform: espressif32; board: esp32dev; framework: arduino)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (6.7.0) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20016.0 (2.0.16)
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 38 compatible libraries
Scanning dependencies...
Dependency Graph
|-- audio-driver @ 0.0.1
|-- audio-tools @ 0.9.8
|-- libhelix @ 0.1.0
|-- SdFat - Adafruit Fork @ 2.2.3
Building in release mode
Compiling .pio\build\esp32dev\src\main.cpp.o
In file included from .pio/libdeps/esp32dev/audio-tools/src/AudioLibs/AudioBoardStream.h:4,
                 from src/main.cpp:9:
.pio/libdeps/esp32dev/audio-tools/src/AudioLibs/I2SCodecStream.h:25:3: error: 'input_device_t' does not name a type; did you mean 'ip_event_t'?
...


I could see that the audio-driver I've got by adding it using the PlatformIO Library Manager took the version 0.0.1 from GitHub. After I've removed this lib and used the direct github link like in your setup I could compile the example. So it looks that Platform IO is not using the latest release (0.0.1 is the only version that the library manager shows)... 

Problem solved and learned a lot... thank you very much... 

now I have to find out more HW details on this board ;-)

Thanks and I hope this could also help others. Looking forward to see more about your excellent work.

Kind regards
Bernhard
